py/src/zenbase/predefined/single_class_classifier/function_generator.py

```diff
@@ -0,0 +1,109 @@
+"""
+This module provides a SingleClassClassifierLMFunctionGenerator for generating language model functions.
+"""
+
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Dict, Optional, Type
+
+from instructor.client import AsyncInstructor, Instructor
+from pydantic import BaseModel
+
+from zenbase.core.managers import ZenbaseTracer
+from zenbase.predefined.base.function_generator import BaseLMFunctionGenerator
+from zenbase.types import LMFunction, LMRequest
+
+
+@dataclass(kw_only=True)
+class SingleClassClassifierLMFunctionGenerator(BaseLMFunctionGenerator):
+    """
+    A generator for creating single-class classifier language model functions.
+    """
+
+    instructor_client: Instructor | AsyncInstructor
```

Comment:
Is there a way to remove dependency on the Instructor client? Ideally a user submits their own LM function with an initial prompt.

Reply:
Would be cool to return the results in an OpenAI compatible kwargs form so the user can consume them however they want

Reply:
we want to make sure we are getting the structured output.